### DISCUSSION TITLE
max-columns in ripgreprc breaks scripts

### DISCUSSION THREAD

**Discussion by anuramat** [BurntSushi/ripgrep#2953] [General] [answered]
wouldn't it be a good idea to make `--max-columns` work kinda like `--color=auto`? it's pretty surprising/astonishing to so get lines cut off in a script.
of course one can just do `--no-config`/`--max-columns=0`...

**Comment by BurntSushi** [comment, marked as answer]
No, I think the current behavior is right. Coloring is a bit different because of ANSI escapes making output effectively unreadable. But max columns is just a normal output setting. What other output settings should act like color? I think it's simpler to not do _too_ much guesswork (which is exactly what `--color=auto` does, guesswork).

**Reply by anuramat** [reply]
thanks, makes sense
